### PR TITLE
Bump packages

### PR DESCRIPTION
**Pull request by cpuguy83**
Bumps k8s deps to v1.24 and some other things involved in making that happen.

### CODE REVIEW

**Review by sargun — APPROVED** [review]

**Comment by cpuguy83** [comment]
Fixed a bunch of things, but still there are some test failures.

**Comment by cpuguy83** [comment]
There we go, all green now.

File: Dockerfile [hunk 1]
```diff
@@ -7,8 +9,24 @@ ARG BUILD_TAGS=""
...
+    --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/golangci-lint \
+    golangci-lint run -v $(if [ -n "${OUT_FORMAT}" ]; then echo "--out-format=${OUT_FORMAT}"; fi)
+
 FROM scratch
 COPY --from=builder /usr/bin/virtual-kubelet /usr/bin/virtual-kubelet
 COPY --from=builder /etc/ssl/certs/ /etc/ssl/certs
 ENTRYPOINT [ "/usr/bin/virtual-kubelet" ]
-CMD [ "--help" ]
+CMD [ "--help" ]
```
Comment (helayoty) [inline Dockerfile:32]: nit: add new line

File: Dockerfile [hunk 2]
```diff
@@ -7,8 +9,24 @@ ARG BUILD_TAGS=""
...
+    go mod download
+COPY . .
+ARG OUT_FORMAT
+RUN \
+    --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/golangci-lint \
+    golangci-lint run -v $(if [ -n "${OUT_FORMAT}" ]; then echo "--out-format=${OUT_FORMAT}"; fi)
+
 FROM scratch
```
Comment (helayoty) [inline Dockerfile:28]: I wish if we could get rid of scratch and use disrtoless instead,

File: Makefile [hunk 3]
```diff
@@ -183,6 +183,10 @@ envtest: kubebuilder_2.3.1_${TEST_OS}_${TEST_ARCH}
...
+
+export GOLANG_CI_LINT_VERSION ?= v1.48.0
+DOCKER_BUILD ?= docker buildx build
+
 .PHONY: lint
-lint: $(gobin_tool)
-	gobin -run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.33.0 run ./...
\ No newline at end of file
+lint:
+	$(DOCKER_BUILD) --target=lint --build-arg GOLANG_CI_LINT_VERSION --build-arg OUT_FORMAT .
```
Comment (helayoty) [inline Makefile:192]: nit: add new line

File: trace/opentelemetry/opentelemetry_test.go [hunk 4]
```diff
@@ -343,22 +346,23 @@ func TestLogf(t *testing.T) {
...
 			spanName:           "test",
 			logLevel:           lFatal,
-			expectedEvents:     []sdktrace.Event{{Name: ""}},
+			expectedEvents:     []trace.Event{{Name: ""}},
 			expectedAttributes: []attribute.KeyValue{},
 		},
 	}
 
 	for _, tt := range testCases {
+		tt := tt
```
Comment (helayoty) [inline trace/opentelemetry/opentelemetry_test.go:365]: Why do we need this?

Comment (cpuguy83) [reply trace/opentelemetry/opentelemetry_test.go:365]: If tests are run in parallel the variable can be overwritten in the loop.

File: trace/opentelemetry/opentelemetry.go [hunk 5]
```diff
@@ -235,26 +235,26 @@ func makeAttribute(key string, val interface{}) (attr attribute.KeyValue) {
 	switch v := val.(type) {
 	case string:
 		return attribute.String(key, v)
-	case []string:
-		return attribute.StringSlice(key, v)
+	// case []string:
```
Comment (helayoty) [inline trace/opentelemetry/opentelemetry.go:238]: Do we need these commented cases? if no, let's remove them instead of commenting.

Comment (cpuguy83) [reply trace/opentelemetry/opentelemetry.go:238]: We do need it... just that the version of otel we had to pull in doesn't support it.
Basically, I'll want to bring this back and just having the comments makes that simpler.

File: node/env_test.go [hunk 6]
```diff
@@ -65,7 +65,7 @@ func wrapE2ETest(ctx context.Context, env *envtest.Environment, f func(context.C
 
 		// The following requires that E2E tests are performed *sequentially*
 		log.L = logger
-		klogv2.SetLogger(logrusr.NewLogger(sl))
+		klogv2.SetLogger(logrusr.New(sl))
```
Comment (helayoty) [inline node/env_test.go:68]: can we use `klog` directly and remove the `klogv2` alias?

**Review by helayoty — COMMENTED** [review]
I added a few comments but overall LGTM. Thanks, @cpuguy83 for doing this work.

**Comment by cpuguy83** [comment]
@caniszczyk Who can I bug about getting the netlify builder updated?
Looks like we got a brown-out due to the builder using Ubuntu 16.04.

```
5:03:46 PM: Build ready to start
5:03:46 PM: ---------------------------------------------------------------------  UNSUPPORTED BUILD IMAGE  The build image for this site uses Ubuntu 16.04 Xenial Xerus, which is no longer supported.  To enable builds for this site, select a later build image at the following link:  https://app.netlify.com/sites/virtual-kubelet/settings/deploys#build-image-selection  For more details, visit the build migration guide:  https://answers.netlify.com/t/please-read-end-of-support-for-xenial-build-image-everything-you-need-to-know/68239  ---------------------------------------------------------------------
```

**Comment by cpuguy83** [comment]
@sargun Still LGTY? I had to make a bunch of changes to make everything work.

File: Dockerfile [hunk 7]
```diff
@@ -7,6 +9,22 @@ ARG BUILD_TAGS=""
...
+    --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    go mod download
+COPY . .
+ARG OUT_FORMAT
+RUN \
+    --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.cache/golangci-lint \
+    golangci-lint run -v $(if [ -n "${OUT_FORMAT}" ]; then echo "--out-format=${OUT_FORMAT}"; fi)
```
Comment (sargun) [inline Dockerfile:26]: Wouldn't this be easier as:

`--out-format=${OUT_FORMAT:-colored-line-number}`

File: node/lifecycle_test.go [hunk 8]
```diff
@@ -28,7 +28,7 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/tools/record"
 	watchutils "k8s.io/client-go/tools/watch"
-	"k8s.io/klog"
+	"k8s.io/klog/v2"
```
Comment (sargun) [inline node/lifecycle_test.go:31]: Might we make the imports of `klog/v2` consistently `klogv2`?

**Review by sargun — APPROVED** [review]

File: node/lifecycle_test.go (hunk 8 above)
Comment (cpuguy83) [reply node/lifecycle_test.go:31]: I think after this change there is no more klog v1 usage.

**Pull request merged**